rust/ql/test/library-tests/variables/Ssa.expected

```diff
@@ -326,7 +331,10 @@ firstRead
 | variables.rs:400:9:400:9 | x | variables.rs:400:9:400:9 | x | variables.rs:406:15:406:15 | x |
 | variables.rs:402:9:402:11 | cap | variables.rs:402:9:402:11 | cap | variables.rs:405:5:405:7 | cap |
 | variables.rs:402:15:404:5 | <captured entry> x | variables.rs:400:9:400:9 | x | variables.rs:403:19:403:19 | x |
+| variables.rs:410:9:410:13 | x | variables.rs:410:13:410:13 | x | variables.rs:416:15:416:15 | x |
 | variables.rs:412:9:412:16 | closure1 | variables.rs:412:9:412:16 | closure1 | variables.rs:415:5:415:12 | closure1 |
+| variables.rs:412:20:414:5 | <captured entry> x | variables.rs:410:13:410:13 | x | variables.rs:413:19:413:19 | x |
+| variables.rs:418:9:418:13 | y | variables.rs:418:13:418:13 | y | variables.rs:424:15:424:15 | y |
```

Comment:
This line illustrates the problem with allowing for captured mutable variables: The test snippet is
```rust
    let mut y = 2; // y [line 418]
    // Captures mutable value by mutable reference
    let mut closure2 = || {
        y = 3; // $ write_access=y
    };
    closure2(); // $ read_access=closure2
    print_i64(y); // $ read_access=y [line 424]
```
It is not true that the value written at line 418 is the one read at line 424, because the value is updated in the call to `closure2`.

One way that we can fix it is to say that mutable captured variables may be read/written via _any_ call, by adding uncertain pseudo reads and writes at all call nodes. In Ruby this is done in the predicates 
`capturedCallRead` and `capturedCallWrite`, respectively.

Reply:
That makes sense and I did not think of that. I'll look at the Ruby approach and try an adapt it.